pkg/endpoint/endpoint.go

```diff
@@ -2323,10 +2318,14 @@ func (e *Endpoint) identityResolutionIsObsolete(myChangeRev int) bool {
 
 // Must be called with e.Mutex NOT held.
 func (e *Endpoint) runLabelsResolver(owner Owner, myChangeRev int) {
-	e.Mutex.Lock()
+	// NOTE: need to figure out if we need e.LockAlive() here
+	// The disco error will be reported by resolve-identity-e.ID controller
+	// which seems to be enough
+	e.UnconditionalLock()
```

Comment:
Unconditional lock is fine here, this is only logging. The main purpose for checking the liveness is to prevent bigger blocks of work that are about to be scheduled.